### PR TITLE
doc/dev/crimson: Improve crimson.rst

### PR DESCRIPTION
**Pull request by anthonyeleven**
doc/dev/crimson: Improve crimson.rst

Improve spelling, wording, grammar, formatting, and clarity.

Signed-off-by: Anthony D'Atri <anthonyeleven@users.noreply.github.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/dev/crimson/crimson.rst [hunk 1]
```diff
@@ -2,90 +2,91 @@
...
-crimson-osd won't instantiate pgs from non-crimson pools.
+The last causes pools to be created by default with the ``crimson`` flag.
+Crimson pools are restricted to operations supported by Crimson.
+``Crimson-osd`` won't instantiate PGs from non-Crimson pools.
 
 Running Crimson
 ===============
 
-As you might expect, crimson is not featurewise on par with its predecessor yet.
+As you might expect, Crimson is not yet on par with its predecessor yet with
```
Comment (zdover23) [inline doc/dev/crimson/crimson.rst:70]: s/is not yet on par with its predecessor yet/is not yet on par with its predecessor/

Or maybe "Crimson does not yet have as extensive a feature set as does ``ceph-osd``."

**Review by zdover23 — APPROVED** [review]
Approved with a suggestion.

File: doc/dev/crimson/crimson.rst [hunk 2]
```diff
@@ -95,36 +96,34 @@ managed by the Seastar framework. They are:
...
-those who are using sysvinit, they can use ``start-stop-daemon`` for daemonizing
-``crimson-osd``. If this is not acceptable, we can whip up a helper utility
-to do the trick.
+a replica of the thread that called `fork()`_. Tackling this problem in Crimson
+would unnecessarily complicate the code.
 
+Since supported GNU/Linux distributions use ``systemd``, which is able to
+daemonize the application, there is no need to daemonize ourselves. 
+Those using sysvinit can use ``start-stop-daemon`` to daemonize ``crimson-osd``.
+If this is not acceptable, that's too damned bad.
```
Comment (zdover23) [inline doc/dev/crimson/crimson.rst:115]: Harsh, but let it stand. I don't really have an informed opinion on the sysvinit/systemd fight.

File: doc/dev/crimson/crimson.rst [hunk 3]
```diff
@@ -140,44 +139,42 @@ over ``20`` will be printed using ``logger::trace()``.
 | >  20   | trace   |
 +---------+---------+
 
-Please note, ``crimson-osd``
-does not send the logging message to specified ``log_file``. It writes
-the logging messages to stdout and/or syslog. Again, this behavior can be
+Please note that ``crimson-osd``
```
Comment (zdover23) [inline doc/dev/crimson/crimson.rst:142]: s/Please//

(Remove "Please".)

File: doc/dev/crimson/crimson.rst [hunk 4]
```diff
@@ -220,37 +217,37 @@ Another SeaStore example::
     --seastore-secondary-devs /dev/sdb \
     --seastore-secondary-devs-type HDD
 
-You could stop the vstart cluster using::
+You may stop this ``vstart`` cluster by running::
```
Comment (zdover23) [inline doc/dev/crimson/crimson.rst:220]: Let's remove modality entirely.

s/You may stop this ``vstart`` cluster by running::/Stop this ``vstart`` cluster by running/

File: doc/dev/crimson/crimson.rst [hunk 5]
```diff
@@ -263,11 +260,11 @@ fio
 ---
 
 ``crimson-store-nbd`` exposes configurable ``FuturizedStore`` internals as an
-NBD server for use with fio.
+NBD server for use with ``fio``.
 
-To use fio to test ``crimson-store-nbd``,
+In order to use ``fio`` to test ``crimson-store-nbd`` perform the below steps.
```
Comment (zdover23) [inline doc/dev/crimson/crimson.rst:265]: s/``crimson-store-nbd``/``crimson-store-nbd``,/

(Add a comma between the clauses.)

File: doc/dev/crimson/crimson.rst [hunk 6]
```diff
@@ -285,7 +282,7 @@ To use fio to test ``crimson-store-nbd``,
       ninja crimson-store-nbd
 
 #. Run the ``crimson-store-nbd`` server with a block device. Please specify
```
Comment (zdover23) [inline doc/dev/crimson/crimson.rst:284]: s/Please//

We don't ever need to ask nicely. They've already come to our documentation for information. (inb4: I know that this "Please" isn't your work.)

File: doc/dev/crimson/crimson.rst [hunk 7]
```diff
@@ -301,16 +298,16 @@ To use fio to test ``crimson-store-nbd``,
...
-   in which,
+   Below are descriptions of these command line arguments:
 
    ``--smp``
-     how many CPU cores are used
+     The number of CPU cores to use (Symmetric MultiProcessor)
 
    ``--mkfs``
-     initialize the device first
+     Initialize the device first
```
Comment (zdover23) [inline doc/dev/crimson/crimson.rst:307]: s/first/first./

(Add a period (if we're making sentences here, which looks likely).)

File: doc/dev/crimson/crimson.rst [hunk 8]
```diff
@@ -337,17 +334,17 @@ To use fio to test ``crimson-store-nbd``,
       [job0]
       offset=0
 
-#. Test the crimson object store using the fio compiled just now
+#. Test the Crimson object store using the custom ``fio`` built just now
```
Comment (zdover23) [inline doc/dev/crimson/crimson.rst:337]: s/object store using/object store, using/

(Add a comma between the two clauses of the sentence.)

File: doc/dev/crimson/crimson.rst [hunk 9]
```diff
@@ -337,17 +334,17 @@ To use fio to test ``crimson-store-nbd``,
...
 
       ./fio nbd.fio
 
 CBT
 ---
-We can use `cbt`_ for performing perf tests::
+We can use `cbt`_ for performance tests::
 
-  $ git checkout master
+  $ git checkout main
```
Comment (zdover23) [inline doc/dev/crimson/crimson.rst:347]: Good looking out.

File: doc/dev/crimson/crimson.rst [hunk 10]
```diff
@@ -459,8 +456,8 @@ so we need to copy and paste the above addresses, then send the EOF by inputting
   seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at /home/kefu/dev/ceph/build/../src/seastar/src/core/app-template.cc:173 (discriminator 5)
   main at /home/kefu/dev/ceph/build/../src/crimson/osd/main.cc:131 (discriminator 1)
 
-Please note, ``seastar-addr2line`` is able to extract the addresses from
-the input, so you can also paste the log messages like::
+Please note that ``seastar-addr2line`` is able to extract addresses from
```
Comment (zdover23) [inline doc/dev/crimson/crimson.rst:459]: s/Please note/Note/

File: doc/dev/crimson/crimson.rst (hunk 2 above)
Comment (anthonyeleven) [reply doc/dev/crimson/crimson.rst:115]: _Crazy People_ reference.  Said fight has clearly been tilting at windmills for years. We had more or less figured out sysvinit, so naturally we had to mess with it.

File: doc/dev/crimson/crimson.rst (hunk 7 above)
Comment (anthonyeleven) [reply doc/dev/crimson/crimson.rst:307]: Added, though the one above is a non-sentence without a period.  With some modern constructs application of the rules of capitalization and punctuation isn't always clear.

File: doc/dev/crimson/crimson.rst [hunk 11]
```diff
@@ -140,44 +139,42 @@ over ``20`` will be printed using ``logger::trace()``.
...
-To facilitate the development of crimson, following options would be handy when
-using ``vstart.sh``,
+The following options aree handy when using ``vstart.sh``,
 
 ``--crimson``
     Start ``crimson-osd`` instead of ``ceph-osd``.
 
 ``--nodaemon``
-    Do not daemonize the service.
+    Do not daemonize the service or coal.
```
Comment (anthonyeleven) [inline doc/dev/crimson/crimson.rst:158]: @zdover23 you missed this one ;)

Comment (zdover23) [reply doc/dev/crimson/crimson.rst]: It's true.

**Pull request merged**